M2/Macaulay2/m2/document.m2

```diff
@@ -1,1429 +1,756 @@
-hasDocumentation = key -> (
-     tag := makeDocumentTag(key,Package=>null);
-     if DocumentTag.PackageName tag === "" then error("key to be documented is exported by no package: ", DocumentTag.FormattedKey tag);
-     pkg := getpkg DocumentTag.PackageName tag;
-     fkey := DocumentTag.FormattedKey tag;
-     null =!= fetchRawDocumentation(pkg,fkey))
+format   DocumentTag := tag -> tag.Format
+package  DocumentTag := tag -> tag.Package
+toString DocumentTag :=
+net      DocumentTag := tag -> concatenate (package tag, " :: ", format tag)
+
+-- FIXME: this is kind of a hack
+toExternalString DocumentTag := tag -> (
+    "new DocumentTag from " | toExternalString {
+	tag.Key, tag.Format, tag.Package})
+
+new DocumentTag from BasicList := (T, t) -> (
+    new DocumentTag from new HashTable from {
+	Key                => t#0,
+	Format             => t#1,
+	symbol Package     => t#2,
+	"RawDocumentation" => if t#?3 then t#3 else null
+	})
+
+DocumentTag ? DocumentTag := (x, y) -> x.Format ? y.Format
+DocumentTag ? String      := (x, y) -> x.Format ? y
+String      ? DocumentTag := (x, y) -> x        ? y.Format
+
+-- helper for parsing " pkg :: key " to ("pkg", "key")
+parseDocumentTag := key -> (
+    segments := separate("^[[:space:]]*|[[:space:]]*::[[:space:]]*|[[:space:]]*$", key);
+    segments  = select(segments, segment -> segment =!= "");
+    -- this is important, because the names of info nodes get extracted from text where
+    -- lines might be wrapped and multiple spaces are reduced to one:
+    if any(segments, segment -> match(" {2,}", segment))
+    then error("expected key to have only single interior spaces:", format key);
+    if      #segments == 0 then error("encountered empty documentation tag: ", format key)
+    else if #segments == 1 then (null,       segments#0)
+    else if #segments == 2 then (segments#0, segments#1)
+    else error("encountered invalid documentation tag: ", format key))
+
+makeDocumentTag' := opts -> key -> (
+    nkey := normalizeDocumentKey(key, opts);
+    verifyKey nkey;
+    fkey := formatDocumentTag nkey;
+    local pkg;
+    (pkg, fkey) = parseDocumentTag fkey;
+    pkg = if class nkey === Symbol -* and package nkey =!= Core *- then package nkey
+    else if opts#Package =!= null then opts#Package else packageKey(key, fkey);
+    if pkg === null then error("makeDocumentTag: package cannot be determined: ", nkey);
+    new DocumentTag from new HashTable from {
+	Key            => if instance(nkey, Symbol) then toString nkey else nkey,
+	Format         => fkey,
+	symbol Package => (
+	    if instance(pkg, Package) then if pkg === Core then "Macaulay2Doc" else pkg#"pkgname" else
+	    if instance(pkg, Symbol) then toString pkg else
+	    if instance(pkg, String) then pkg else ""),
+	"RawDocumentation" => null
+	})
+
+makeDocumentTag = method(Dispatch => Thing, Options => { Package => null })
```

Comment:
Should be fixed in the next commit.

Reply:
Actually, I realized why this is happening. See this "kludge" line:
https://github.com/Macaulay2/M2/blob/2eb5731e107c64c5574228207487da6d2edb5e48/M2/Macaulay2/m2/document.m2#L140-L145
Removing that line would fix this, but it would also break Schubert2. I'm not sure what's the best fix for this.

Reply:
Good catch!  What if I just fix the documentation for Schubert2 so it doesn't try to document that?

Reply:
I think that's bad for many reasons. Packages should be able to build on top of each other and document their new behavior. Even if you fix this one, the Complexes package will soon run into the same issue multiplied by 10.

Reply:
That's true, but it will be more work and better done later.  Right now we have something to fix.

Reply:
What do we have to fix? This is for all intents and purposes harmless. Notice that `help makeDocumentTag (res, Module)` still produces the right node.

If you insist, we can remove the assertion that documentation keys belong to the same package for now, then we can safely remove this kludge.

Reply:
Also, notice that this used to be broken too:
https://faculty.math.illinois.edu/Macaulay2/doc/Macaulay2-1.16/share/doc/Macaulay2/Macaulay2Doc/html/_make__Document__Tag.html

And besides, I'm pretty sure we can safely unexport makeDocumentTag.

I don't think this is an issue.

Reply:
> What do we have to fix? This is for all intents and purposes harmless. Notice that `help makeDocumentTag (res, Module)` still produces the right node.
> 

It seems you have fixed something, for today I get this:
```m2
i7 : makeDocumentTag ( res, Module)

o7 = Macaulay2Doc :: resolution(Module)

o7 : DocumentTag
```